src/main/java/Kratos.java

```diff
@@ -0,0 +1,187 @@
+        System.out.printf("         %s%n",  tasksList[taskNumber].toString());
+        System.out.println(LINE);
+    }
+
+    // Method to display tasks
+    public static void displayTasks(int count) {
+        System.out.println(LINE);
+        System.out.println("    Your list of Tasks");
+        for (int i = 0; i < count; i++) {
+            System.out.printf("     %d. %s%n", i + 1, tasksList[i].toString());
+        }
+        System.out.println(LINE);
+    }
+
+    // Main method
+    public static void main(String[] args) {
+        greet();
+        String userInput;
+        Scanner in = new Scanner(System.in);
+        while (true) {
+            userInput = in.nextLine();
+            if (userInput.equals("bye")) {
+                break;
+            } else if (userInput.equals("list")) {
+                displayTasks(count);
+            } else if (userInput.startsWith("mark") || userInput.startsWith("unmark")) {
+                int taskNumber = Integer.parseInt(userInput.split(" ")[1]); //Extract task number
+                displayMarking(taskNumber - 1, userInput.split(" ")[0]);
+            } else {
+                try {
+                    if (userInput.startsWith("deadline")) {
+                        addDeadline(userInput);
+                    }
+                    else if (userInput.startsWith("todo")) {
+                        addTodo(userInput);
+
+                    }
+                    else if (userInput.startsWith("event")) {
+                        addEvent(userInput);
+                    }
+                    else {
+                        throw new IllegalArgumentException();
+                    }
+                }
+                catch (Exception e) {
+                    KratosException.handleException(e, userInput);
+                }
+            }
+        }
+        end();
+    }
+
+    private static void addEvent(String userInput) {
+        // Split the string by "/by" to separate the deadline and date
+        String[] eventParts = userInput.split("/from");
+
+        // Extract the deadline, action, and item
+        String eventAndName = eventParts[0].trim();  // "event project meeting"
+        String date = eventParts[1].trim();  // "Mon 2pm /to 4pm"
+
+        // Split the deadline into action and item
+        String[] event = eventAndName.split("\\s+", 2);
+        String action = event[1]; // project Meeting
+
+        String[] timeline = date.split("/to");
+
+        tasksList[count] = new Event(action, timeline[0].trim(), timeline[1].trim());
+        count++;
+        System.out.println(LINE);
+        System.out.println("Event recorded. Destiny's hourglass turns.\n" +
+                "What now? Seize control or be swept by its sands?");
+        System.out.println(LINE);
+    }
+
+    private static void addTodo(String userInput) {
+        // Split line into action and type
+        String[] actionAndType = userInput.split("\\s+", 2);
```

Comment:
Plural form should be used on names representing a collection of objects. perhaps might want to change to another suitable variable name :] (e.g. userInputLists/userInputArgumentLists/toDoParts)?